### ISSUE TITLE
LA-VAE & LA-AE

### ISSUE THREAD

**Issue by hulne** [WinfredGe/T2S#2]
Thank you to the author for the great work.

However, it appears that the code for LA-VAE actually uses a classic AutoEncoder, and does not apply the KL loss to constrain the latent space as in a VAE.  
```
if mode == 'train':
    optimizer.zero_grad()
    z, before = self.encoder(batch)
    data_recon, after = self.decoder(z, length=batch.shape[-1])
    recon_error = F.mse_loss(data_recon, batch)
    cross_loss = F.mse_loss(before, after)
    loss = recon_error + cross_loss
    loss.backward()
    optimizer.step()
```
May I ask why it is named LA-VAE instead of LA-AE? Also, can this design ensure good generation quality and meaningful reverse diffusion? From what I have seen in the original VAE-DDPM paper, if it is changed to AE-DDPM, it seems there is no theoretical guarantee.

**Comment by KaigeZheng** [comment]
I have the same question. (Also, the file and class names are still ‘vqvae’ — maybe you forgot to change them?)

**Comment by KaigeZheng** [comment]
> Thank you to the author for the great work.
> 
> However, it appears that the code for LA-VAE actually uses a classic AutoEncoder, and does not apply the KL loss to constrain the latent space as in a VAE.
> 
> ```
> if mode == 'train':
>     optimizer.zero_grad()
>     z, before = self.encoder(batch)
>     data_recon, after = self.decoder(z, length=batch.shape[-1])
>     recon_error = F.mse_loss(data_recon, batch)
>     cross_loss = F.mse_loss(before, after)
>     loss = recon_error + cross_loss
>     loss.backward()
>     optimizer.step()
> ```
> 
> May I ask why it is named LA-VAE instead of LA-AE? Also, can this design ensure good generation quality and meaningful reverse diffusion? From what I have seen in the original VAE-DDPM paper, if it is changed to AE-DDPM, it seems there is no theoretical guarantee.

It appears that this VAE seems to have been modified from the source code of this reference repository (https://github.com/SaberaTalukder/TOTEM).

**Comment by WinfredGe** [comment]
Yes, you are right. I apologize for the inaccurate naming. The current implementation is actually a simple AE rather than a VAE, since it does not include the KL divergence term for latent space regularization. Thank you for pointing this out.
